### DISCUSSION TITLE
events data not getting collected in nextjs 13

### DISCUSSION THREAD

**Discussion by nkilm** [umami-software/umami#1807] [Q&A]
Hello, I've followed all the instructions given the [docs](https://umami.is/docs/getting-started) on adding umami to nextjs. But still the data is not getting collected. I am using [planetscale](https://planetscale.com/) to store all the umami data. 

website - [nikhilmohite.com](https://nikhilmohite.com)

The umami script tag is also present on the website, database connection is also successful. But still it's not working. 

![image](https://user-images.githubusercontent.com/79012023/222779836-fc025605-6ac4-45ff-8fc5-e461359d8241.png)

**No data related to events nor any activity is getting collected.**

planetscale database

![image](https://user-images.githubusercontent.com/79012023/222780476-ee5451c9-646b-4217-82a2-810b344fb87a.png)

**Comment by Milkywayrules** [comment]
This is how I implement Umami using next.js 13 (without app folder) inside `pages/index.tsx` for our landing page:

![image](https://user-images.githubusercontent.com/22197973/225238035-26524e8b-3364-4c2b-8069-06bde229fead.png)

I've tried put that code inside `_document.tsx` but it doesn't work (at least for me, maybe). If you have multiple pages on your website, you better put that code inside your custom `Layout` component that wrap everything inside `_app.tsx`

Hope it helps, cheers!

**Reply by nkilm** [reply]
@Milkywayrules Thank you for the suggestion!😀 Now the data is getting collected. Apparently the problem was with database. I was using [supabase](https://supabase.com/) previously, now it's working fine after switching to [planetscale](https://planetscale.com/).

I have included the umami script in `_app.tsx`. 

```tsx
	  {/* umami analytics */}
      {/* Reference - https://umami.is/docs/collect-data */}
      {process.env.NEXT_PUBLIC_UMAMI_WEBSITE_ID && (
        <Script
          data-website-id={process.env.NEXT_PUBLIC_UMAMI_WEBSITE_ID || ""}
          src="https://umami-coral-nu.vercel.app/umami.js"
          strategy="lazyOnload"
        />
      )}
```
Also I notice that you are using the native script tag, it is recommended to replace it with **Script** from [next/script](https://nextjs.org/docs/basic-features/script).

**Reply by XahidEx** [reply]
Bro, Were you able to transfer your data from Supabase to planetscale? If you did how?

**Reply by nkilm** [reply]
Hey @XahidEx! I am not using supabase, I am using planetscale. Unfortunately I had to reinstall everything again with fresh database. All data related to v1 is gone:(
